src/common/ipc/ipc.ts

```diff
@@ -71,34 +68,18 @@ export function broadcastMessage(channel: string, ...args: any[]) {
     });
 }
 
-export function subscribeToBroadcast(channel: string, listener: (...args: any[]) => any) {
-  if (ipcRenderer) {
-    ipcRenderer.on(channel, listener);
-  } else if (ipcMain) {
-    ipcMain.on(channel, listener);
-  }
+export function ipcMainOn(channel: string, listener: (event: Electron.IpcMainEvent, ...args: any[]) => any): Disposer {
+  ipcMain.on(channel, listener);
 
-  return listener;
+  return () => ipcMain.off(channel, listener);
 }
 
-export function unsubscribeFromBroadcast(channel: string, listener: (...args: any[]) => any) {
-  if (ipcRenderer) {
-    ipcRenderer.off(channel, listener);
-  } else if (ipcMain) {
-    ipcMain.off(channel, listener);
-  }
-}
+export function ipcRendererOn(channel: string, listener: (event: Electron.IpcRendererEvent, ...args: any[]) => any): Disposer {
+  ipcRenderer.on(channel, listener);
```

Comment:
We should still check if `ipcRenderer`, etc.  is valid in these functions?

Reply:
Oh do you mean for tests and all? Currently the checks are usually done in the preceding scopes.

Reply:
If this function is exported to the extensions api it should be bulletproofed. Or regardless. Just because it's called `ipcRendererOn` doesn't mean it will be called from the right process.

Reply:
This function is not directly exposed to the extensions API. See #2775